### PR TITLE
Windows file path clean

### LINKED ISSUE #71: [BUG] Not opening files from telescope preview in folders outside of home in Windows
**Please confirm**
- [x] I am running the latest version of this plugin

**Describe the bug**
When I attempt to open a note while in a directory that isn't `~/zettelkasten` using one of the commands from the panel, it will create a new file in my current working directory named "c".

**To Reproduce**
1. Start nvim
2. Go to a directory that isn't the home directory as listed in the config in init.vim
3. Search for a file through one of the keyboard mappings or call a command (`<leader>zf` or `<leader>zg` are what I first used to replicate the problem)
4. Select a Zettel that appears when searching in telescope.

**Expected behavior**
I expected the file that I selected in my zettelkasten to open.

**Screenshots**
I am on my phone at work at the moment, but when I get home I will add screenshots of what it is doing

**Operating system (please complete the following information):**
 - OS: Windows 10

**Additional context**
This problem occurs even if I use an init.vim with only the plugins required for this one (the minimum init file you can create and still have this plugin work). Again, at work so I'll post it here when I get the chance tonight.

### PR DESCRIPTION
**Pull request by AuLaSW**
## Proposed change

This PR fixes #71. I've added a function called `CleanPath` to `telekasten.lua` that takes the incoming `home` path (either default or set by the user in their config) and checks if it is a Windows path. If not, it returns the original path. If it is, it substitutes the path with a substring of the path that begins at the first folder delimiter (`\\` in Windows, `/` in Unix/Linux).

This change allows Windows users to use `vim.fn.expand('~\zettelkasten')` like in the default config (or any variation of the sort) without having to worry about `C:` messing up the file path.

## Type of change

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Code quality improvements to existing code or addition of tests
- [ ] Documentation update

## Additional information

- This PR fixes or closes issue: fixes #71
- This PR is related to issue: #71

## Checklist

- [x] I am running the **latest** version of the plugin.
- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] The code has been formatted using Stylua (a `.stylua.toml` file is provided)
- [x] The code has been checked with luacheck (a `.luacheckrc` file is provided)
- [x] The `README.md` has been updated according to this change.
- [x] The `doc/telekasten.txt` helpfile has been updated according to this change.

### CODE REVIEW

File: lua/telekasten.lua [hunk 1]
```diff
@@ -2487,6 +2512,8 @@ local function Setup(cfg)
         -- merge everything but calendar opts
         -- they will be merged later
         if k ~= "calendar_opts" then
+            if k == "home" then
+                v = CleanPath(v)
```
Comment (renerocksai) [inline lua/telekasten.lua:2516]: There seems to be no `end` for the if. Are you sure you tested this? Or is GitHub playing tricks on me?

File: lua/telekasten.lua [hunk 2]
```diff
@@ -23,10 +23,35 @@ local Path = require("plenary.path")
...
+    
+    -- Returns cleaned path
+    return path
+end
+
 -- ----------------------------------------------------------------------------
 -- DEFAULT CONFIG
 -- ----------------------------------------------------------------------------
 local home = vim.fn.expand("~/zettelkasten")
+home = CleanPath(home)
```
Comment (renerocksai) [inline lua/telekasten.lua:54]: OK, first I thought this is unnecessary - but... Why not :smile: ? If someone forgets to set it.

**Review by renerocksai — CHANGES_REQUESTED** [review]
OK, before merging, please also update the help file (`doc/telekasten.txt`)

And also please check for the missing end. No end for the if makes me think you didn't test this. I will double check with the file view, though.

File: lua/telekasten.lua (hunk 1 above)
Comment (AuLaSW) [reply lua/telekasten.lua:2516]: I looked, and what happened was that I had fixed it on my laptop, but had not committed and pushed to GitHub. Got onto my work computer and did a pull request and didn't think to check. This is my first time doing a PR so I got a bit excited and didn't think too clearly. I've fixed it though!

**Comment by AuLaSW** [comment]
@renerocksai I think I fixed the doc file how you wanted? Also I apologize for not checking before I pushed. I'll make sure to check next time.

There are still a lot of issues with paths in Windows. I know you didn't originally design this to work with Windows, so I don't expect it to work too well, lol. I'm still figuring out how to fix them so if I do manage to do that I'll let you know and send a PR.

Thank you for taking the time to look over this. I've never done something like this before and didn't think to check before I sent it in, so I've already learned a decent bit just from this one experience.

**Comment by renerocksai** [comment]
Great, doc is looking good now (from what I've seen). The automated `luacheck` should also catch syntax errors...

Oh, I just ran it and now it fails the `stylua` check. It complains `(1,1)` missing a space: `(1, 1)`. Can you fix that? Then we'd have it all contained within this PR.  So for that you don't even need to install stylua.

**Comment by renerocksai** [comment]
Please also check the luacheck error. Usually, and that's why it's part of the PR template, I encourage people to run both stylua and luacheck (config files are provided in the repo, so you just need to run the 2 commands) to catch stuff like this.

**Pull request merged**